pandas/core/ops.py

```diff
@@ -745,24 +740,19 @@ def na_op(x, y):
         elif is_categorical_dtype(y) and not is_scalar(y):
             return op(y, x)
 
-        if is_object_dtype(x.dtype):
+        elif is_object_dtype(x.dtype):
             result = _comp_method_OBJECT_ARRAY(op, x, y)
+
+        elif is_datetimelike_v_numeric(x, y):
+            raise TypeError("invalid type comparison")
+
         else:
 
             # we want to compare like types
             # we only want to convert to integer like if
             # we are not NotImplemented, otherwise
```

Comment:
comment no longer relevant?

Reply:
The comment may be more verbose than it needs to be, but is still relevant.  Once #19301 goes in we can dispatch to DTI and TDI and the needs_i8_conversion block below can be simplified quite a bit.